### ISSUE TITLE
Add --skipClient and --skipServer flags to generator

### ISSUE THREAD

**Issue by deepu105** [jhipster/generator-jhipster#2582]
There has been many request for a similar feature by different people earlier. but then we didnt have enough use cases for this

I propose to add two non intrusive flags to the main generator so that below use cases can be supported

1.  A module can compose using the main generators to create a Jhipster server side templates and it can provide its own client side, this will make possible to create a JHipster app with a React or Ember front end instead of Angular or someone can simple provide an angular material UI instead of bootstrap, the possibilities are endless 
2. A module can compose the generator and create a Angular JS front end but a different backend config than ours
3. Once we go micro services, there will be a need to create a backend only application and people can easily do `yo jhipster --skipClient`` and get that done
4. This will make our generator more compatible to be used by other generators or API

adding this will not break anything and will not be an issue for any normal users, it would require only few condition checks once we modularize the generator as per https://github.com/jhipster/generator-jhipster/issues/2581, this option can be recorded in the yo-rc file to be read by entity generator as well.

**Comment by jdubois** [comment]
Yes, I fully agree with this idea, then maybe it shouldn't be a flag but a new question. Can you wait a couple of weeks, until the microservices stuff is more mature?

I'm thinking we could have a the beginning a new question:

What type of application would you like to generate?

- Normal JHipster application (monolithic)
- Cloud-enabled JHipster application (registers in Eureka, security is done in an external OAuth2 server)
- Cloud-enabled back-end JHipster application (same as above, but no UI)

That would be less options in the end, but the idea is to guide the users, so it's easier for them to use the generator.

Then I have an unanswered question yet: should we allow to create a "Cloud-enabled back-end JHipster application" with and UI that is in another application (in our API Gateway)? That would be awesome, right? But it could be difficult to generate.

**Comment by deepu105** [comment]
@jdubois yes once the microservice is out it might be good to have that question, my requirement above is more for other generators and modules to compose using Jhipster easily for that purpose flags are more appropriate easy to pass when composing.

Once we modularize the main generator properly this would require only very few if checks overall and in my opinion this can co exist along side the questions you suggest which is more for normal users and not for modules, this flag will be more for modules and advanced users than for normal users

**Comment by jdubois** [comment]
Yes it's the same idea anyway -> OK for me

**Comment by deepu105** [comment]
Great, ill first do a PR to restructure the main generator

**Comment by gmarziou** [comment]
Hi @deepu105 

I tried --skip-client option, it works great, I have few remarks:
- as it is easy to make typos in options names, would it be possible to raise an error if option is not recognized? Maybe this is an option of yo?
- the README.md should not have links to node and other JS tools at the bottom

When going to root path (http://localhost:8080), I get an error which seems to indicate an issue with i18n properties, I will check.

~~~
??error.title_en_US??

??error.subtitle_en_US??
??error.status_en_US?? Not Found (Not Found)
??error.message_en_US?? No message available
~~~

I got swagger page once and then it was no longer working, I'll have a closer look.
